backend/app/component_handler_test.go

```diff
@@ -0,0 +1,310 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+
+	"github.com/codescalersinternships/Flyspray/models"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateComponent(t *testing.T) {
+
+	tempDir := t.TempDir()
+
+	dbPath := filepath.Join(tempDir, "testing.db")
+
+	app, err := NewApp(dbPath)
+	assert.Nil(t, err)
+
+	app.router = gin.Default()
+	app.router.Use(func(c *gin.Context) {
+		c.Set("user_id", "1")
+		c.Next()
+	})
+	app.router.POST("/component", WrapFunc(app.createComponent))
+
+	t.Run("Success", func(t *testing.T) {
+		componentInput := createComponentInput{
+			ProjectID: "1",
+			Name:      "Test Component",
+		}
+		payload, err := json.Marshal(componentInput)
+		if err != nil {
+			t.Fatal("failed to marshal comment payload")
+		}
+		req, err := http.NewRequest("POST", "/component", bytes.NewBuffer(payload))
+		assert.Nil(t, err)
+
+		req.Header.Set("Content-Type", "application/json")
+
+		w := httptest.NewRecorder()
+
+		app.router.ServeHTTP(w, req)
+
+		assert.Equal(t, http.StatusCreated, w.Code)
+
+	})
+
+	t.Run("Bad Request", func(t *testing.T) {
+
+		componentInput := createComponentInput{
+			ProjectID: "1",
+			Name:      "Test Component",
+		}
+		payload, err := json.Marshal(componentInput)
+		if err != nil {
+			t.Fatal("failed to marshal comment payload")
+		}
+
+		req, err := http.NewRequest("POST", "/component", bytes.NewBuffer(payload))
+		assert.Nil(t, err)
+
+		req.Header.Set("Content-Type", "application/json")
+
+		w := httptest.NewRecorder()
+
+		app.router.ServeHTTP(w, req)
+
+		assert.Equal(t, http.StatusBadRequest, w.Code)
+	})
+
+	t.Run("Bad Request", func(t *testing.T) {
```

Comment:
both tests are `Bad Request`. add more details to names so they are different. Like why they are bad requests